### PR TITLE
Fix wrong behaviour of SGIO::progress

### PR DESCRIPTION
**Pull request by geektoni**
See #3509.

**SG_PROGRESS current behaviour**:

previous_behaviour

**SG_PROGRESS patch behaviour**:

new_behaviour

### CODE REVIEW

File: src/shogun/io/SGIO.cpp [hunk 1]
```diff
@@ -181,8 +181,18 @@ void SGIO::progress(
 		v=CMath::clamp(v,1e-5,100.0);
 		last_progress = v-1e-6;
 
-		if ((v!=100.0) && (runtime - last_progress_time<0.5))
+		if ((v!=100.0) && (runtime - last_progress_time<0.5)) {
```
Comment (karlnapf) [inline src/shogun/io/SGIO.cpp]: style: we do { in a new line

File: src/shogun/io/SGIO.cpp [hunk 2]
```diff
@@ -181,8 +181,18 @@ void SGIO::progress(
...
+
+			// This is made to display correctly the percentage
+			// if the algorithm execution is too fast
+			if (current_val >= max_val-1){
```
Comment (karlnapf) [inline src/shogun/io/SGIO.cpp]: style

**Comment by geektoni** [comment]
Style fixed.

**Comment by karlnapf** [comment]
Do we actually have a unit test for this thing? Just to ensure nothing crashes?

**Comment by geektoni** [comment]
As far as I know, the entire SGIO has no unit tests, which means no, there is no unit test to ensure the correctness of the code. Shall I spend some time creating one just to be sure it works as before?

**Comment by karlnapf** [comment]
I think a test that at least executes the stuff you added would be good. Correctness is tricky as it just prints out. Not sure it is worth the hassle to check this. But making sure it doesnt crash on corner cases would be useful definitely :)

**Comment by karlnapf** [comment]
This one is fine to merge from my side
